lib/gds_api/saved_page_api.rb

```diff
@@ -0,0 +1,47 @@
+# Adapter for the Saved Page API
+# Authenticated with Account API
+#
+# @see https://github.com/alphagov/account-api
+# @api documented
+
+class GdsApi::SavedPageApi < GdsApi::AccountApi
```

Comment:
I still find this a bit weird, as the subclassing means that `GdsApi::SavedPageApi` has a *superset* of the methods of the `GdsApi::AccountApi`.

If you can always replace `GdsApi::AccountApi` with `GdsApi::SavedPageApi`, why would you ever use `GdsApi::AccountApi`?

Reply:
Actually, having realised that now, I think this should be a separate class *only if* the only methods it exposes are saved page methods.  So right now, I don't think this should be a separate class.

Reply:
I guess I was thinking if the `AccountAPI` was the core API that covered authentication etc.
But then feature specific authenticated APIs were separate you could bring in a specific set methods for a task at hand.

So in that case we might have;
`AccountApi`
`EmailSubsriptionApi`
`SavedPageApi`

I'm not sure where attribute would live in there, they feel kind of integral to the AccountApi 🤔 

otherwise the AccountApi just starts getting huge as we bolt on more and more bits?

Reply:
> otherwise the AccountApi just starts getting huge as we bolt on more and more bits?

But we don't have that problem *yet* - the pattern in gds-api-adapters at the moment is to have one class per app, and even much bigger APIs than ours (like publishing-api or email-alert-api) follow that pattern.

Splitting things up when we have only 12 endpoints feels like a premature optimisation not worth the inconsistency.